website/docs/index.html.markdown

```diff
@@ -37,5 +39,11 @@ The following arguments are supported:
 * `hostname` - (Optional) The Terraform Enterprise hostname to connect to.
   Defaults to `app.terraform.io`.
 * `token` - (Optional) The token used to authenticate with Terraform Enterprise.
-	We recommend omitting the token which can be set as `credentials` in the
+  If omitted, the provider will use a token from the `credentials` block in the
   [CLI config file](/docs/commands/cli-config.html#credentials).
+
+    -> **Note:** The CLI config file is only for use with command-line Terraform
+    runs. When using a Terraform Enterprise workspace to manage resources with
+    this provider, you must explicitly set `token` in the provider configuration
+    block. We recommend using an input variable for the token and marking the
+    corresponding variable in the Terraform Enterprise workspace as sensitive.
```

Comment:
I would make this a bit more explicit. So when using this in your terminal, we strongly recommend to use a token configured in the CLI config file. And only when this provider is being executed in TFE, we recommend you use an input variable to set the `token` field with a token that has the required rights.